src/inmanta/ast/statements/generator.py

```diff
@@ -300,6 +324,9 @@ def requires_emit(self, resolver: Resolver, queue: QueueScheduler) -> Dict[objec
         direct = [x for x in self._direct_attributes.items()]
 
         direct_requires = {rk: rv for (k, v) in direct for (rk, rv) in v.requires_emit(resolver, queue).items()}
+        direct_requires.update(
```

Comment:
this will make all kwargs dicts required even if there are no required_kwargs 

This may be a problem for figuring out the correct execution order that I would like to avoid

Reply:
I agree this is not ideal. However, I believe it is necessary. It's impossible to know the keys of the dicts without waiting for the dict (and in the current CreateDict implementation, for it's values). Therefore at this point we have no way of knowing if there are any index values in there, so we need to wait for all dicts.

Reply:
unless all attributes of all indices are already in the list of named arguments, then they can at most contain (wrong) duplicates.

Reply:
You can defer this to another ticket,
also lazy execution of dicts can be another ticket.

Reply:
#1711, #1712